### DISCUSSION TITLE
Read-Only Tasks

### DISCUSSION THREAD

**Discussion by cwchristerw** [bitfireAT/icsx5#30] [Ideas]
I would like to see feature to get every VTODO as read-only tasks to OpenTasks app.

**Comment by AlexD97** [comment]
Is there manybe any other app that can retrieve the todos from ICSx5?

**Reply by rfc2822** [reply]
Because ICSx⁵ uses the [same library](https://github.com/bitfireAT/ical4android), the same tasks apps would be supported: jtx Board, OpenTasks, tasks.org.

However at the moment there's no support for that.

Do you have a use case for tasks in a Webcal? Maybe even a public Webcal?

**Reply by AlexD97** [reply]
I have a local .ics-file (created from Emacs org-mode files) where calendar events and todos are contained. However, with ICSx5 only the calendar events are exported. I would find it very useful if I could also see my todos in a phone app.

**Reply by webysther** [reply]
> Because ICSx⁵ uses the [same library](https://github.com/bitfireAT/ical4android), the same tasks apps would be supported: jtx Board, OpenTasks, tasks.org.
> 
> However at the moment there's no support for that.
> 
> Do you have a use case for tasks in a Webcal? Maybe even a public Webcal?

My case, I have a list of tasks for house cleaning in nextcloud, the tasks normally show in calendar in the case of nextcloud the ICS is exported with VTODO, which don't show any data because the project look for VEVENT only.

The ICS: [limpeza-2025-07-31.zip](https://github.com/user-attachments/files/21524705/limpeza-2025-07-31.zip)

<img width="1370" height="863" alt="image" src="https://github.com/user-attachments/assets/0cef671b-15b7-4b9d-9d5b-9348a88dcf47" />

All this is only to be able to show my events and task in a calendar view, in the opensource space the only option is nextcloud and for android don't have none, the acalendar don't support tasks.org when using with DAV direct which don't work well because duplicate the tasks list in the app.

**Comment by cwchristerw** [comment]
@rfc2822 I have two phones for different purposes. Due to security reason I can't use my Nextcloud directly from my social media phone. I have created code that will import data from Nextcloud with ics (using [sabre/vobject](https://sabre.io/vobject))  and then it will redact information from events (VEVENT) and currently removes tasks (VTODO).

In my previous setup I used SOGO as Groupware. I had two or more accounts in server that handled different security levels. I had account for Public calendar and tasks. In my new system (Nextcloud) I want to have one account (using SSO nowadays) and use external script to redact sensitive information.

I could probably create system with sabre/dav, but it could take longer to implement. I would use DAVx5 with read only.

**Reply by cwchristerw** [reply]
Instead of removing VTODOs from public calendar. I would use this feature to read incoming (public) tasks and then I would tick them from my other phone that has permissions to Nextcloud.

**Reply by cwchristerw** [reply]
Also code I use is combining multiple calendars to one :smile:

**Comment by diggy128** [comment]
Any update on this?
I have my events exported through ICS and displayed in the [Todo Agenda](https://f-droid.org/en/packages/org.andstatus.todoagenda/) in my home screen.
I also have another ICS that exports VTODOs. Now they are ignored. It would be incredible to have the VTODOs also available.
